browser-extension/mv2/src/background/background.js

```diff
@@ -264,13 +269,50 @@ BG.Methods.modifyHeaders = function (originalHeaders, headersTarget, details) {
     ruleType,
     rulePairs,
     rulePair,
-    isRuleApplied = false,
+    hasModifiedHeaders = false,
     modifications,
     modification,
     url = details.url,
     mainFrameUrl = BG.Methods.getMainFrameUrl(details),
     enabledRules = BG.Methods.getEnabledRules();
 
+  if (headersTarget === RQ.HEADERS_TARGET.REQUEST) {
```

Comment:
We should add a  comment and  link to the Github ticket so that we can get the context of this code.

Reply:
Okay